### PR TITLE
infoschema: add the SchemaNameByTableID API

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

This PR introduced a new API to query the database name using the table ID. This API will be used in the new priority queue implementation. 
Because we want to avoid fetching table information as much as possible. So we need to add this new API to achieve it.

Check https://github.com/pingcap/tidb/pull/55889/files#r1766302195 out to see how we use it in the new priority queue.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/infoschema/infoschema.go [hunk 1]
```diff
@@ -308,6 +308,26 @@ func (is *infoSchema) TableByID(_ stdctx.Context, id int64) (val table.Table, ok
...
+	if !tableIDIsValid(tableID) {
+		return
+	}
+
+	slice := is.sortedTablesBuckets[tableBucketIdx(tableID)]
+	idx := slice.searchTable(tableID)
+	if idx == -1 {
+		return
+	}
+
```
Comment (lance6716) [inline pkg/infoschema/infoschema.go:321]: can reuse `TableByID`

File: pkg/infoschema/infoschema_v2.go [hunk 2]
```diff
@@ -646,6 +646,25 @@ func (is *infoschemaV2) TableByID(ctx context.Context, id int64) (val table.Tabl
 	return ret, true
 }
 
+func (is *infoschemaV2) SchemaNameByTableID(tableID int64) (schemaName pmodel.CIStr, ok bool) {
+	if !tableIDIsValid(tableID) {
```
Comment (lance6716) [inline pkg/infoschema/infoschema_v2.go:650]: also extract a `tableItemByID` from `TableByID` and reuse it.

**Review by tiancaiamao — APPROVED** [review]
This API can be implemented by combine TableByID and SchemaByID:

```
t := TableByID(tid)
d := SchemaByID(t.Meta().DBID)
```

If I write the code, I would prefer **keeping the interface minimal and composible**,
And provide such a function:

```
package infoschema

func SchemaNameByTableID(is infoschema.InfoSchema, tableID int64) (schemaName pmodel.CIStr, ok bool) {
	// If is implements SchemaNameByTableID, use the native implementation for speed.
	if fn, ok := is.(interface{SchemaNameByTableID(tableID int64)}); ok {
		return fn(tableID)
	}

	// Otherwise choose the normal code path.
	tbInfo := is.TableInfoByID(tid)
	dbInfo := is.SchemaByID(tbInfo.DBID)
	return dbInfo.Name, ok
}
```

and use function `infoschema.SchemaNameByTableID`  in the callers.

Anyway, that's just some personal coding taste.
Please address comment from others, rest LGTM

File: pkg/infoschema/infoschema.go (hunk 1 above)
Comment (Rustin170506) [reply pkg/infoschema/infoschema.go]: Great suggestion. I forgot about it.

File: pkg/infoschema/infoschema_v2.go (hunk 2 above)
Comment (Rustin170506) [reply pkg/infoschema/infoschema_v2.go:659]: Updated.

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by lance6716 — APPROVED** [review]

File: pkg/infoschema/infoschema.go [hunk 3]
```diff
@@ -308,6 +308,19 @@ func (is *infoSchema) TableByID(_ stdctx.Context, id int64) (val table.Table, ok
 	return slice[idx], true
 }
 
+func (is *infoSchema) SchemaNameByTableID(tableID int64) (schemaName pmodel.CIStr, ok bool) {
+	tbl, ok := is.TableByID(stdctx.Background(), tableID)
```
Comment (GMHDBJD) [inline pkg/infoschema/infoschema.go:312]: how about pass the ctx instead of the background ctx

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: pkg/infoschema/infoschema.go (hunk 3 above)
Comment (Rustin170506) [reply pkg/infoschema/infoschema.go:312]: We don't really use that context.

**Comment by Rustin170506** [comment]
/hold

We need to consider the partition ID as well.

**Comment by Rustin170506** [comment]
/unhold

Using table id is enough.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by Rustin170506** [comment]
@ti-chi-bot Hi

**Comment by wuhuizuo** [comment]
> @ti-chi-bot Hi

/ping

**Pull request merged**